src/datasets/load.py

```diff
@@ -632,14 +634,14 @@ def get_module(self) -> DatasetModule:
             base_path=base_path,
             allowed_extensions=ALL_ALLOWED_EXTENSIONS,
         )
-        module_names = {
-            key: infer_module_for_data_files(data_files_list) for key, data_files_list in data_files.items()
+        split_modules = {
+            split: infer_module_for_data_files(data_files_list) for split, data_files_list in data_files.items()
         }
-        if len(set(list(zip(*module_names.values()))[0])) > 1:
-            raise ValueError(f"Couldn't infer the same data file format for all splits. Got {module_names}")
-        module_name, builder_kwargs = next(iter(module_names.values()))
+        module_name, builder_kwargs = next(iter(split_modules.values()))
```

Comment:
Maybe check that i split_modules is not empty ?

Reply:
Thanks for your review, @lhoestq.

I think it can only be empty if the user passes `data_files={}`, otherwise there are 2 options: either it is not empty or an exception is raised.
- `split_modules` is derived from `data_files`, which is instance of `DataFilesDict.from_local_or_remote` with `patterns`
- `patterns` is derived either from `sanitize_patterns` or `get_data_patterns_locally`
  -  `sanitize_patterns` can only return an empty dict if the user passes `data_files={}`
  - `get_data_patterns_locally` can only return a non-empty dict or raise a `EmptyDatasetError`

I think the validation of  `data_files={}` should be elsewhere though. What do you think?

Reply:
Maybe changing?
```python
sanitize_patterns(self.data_files) if self.data_files is not None else get_data_patterns_locally(base_path)
```
to
```python
sanitize_patterns(self.data_files) if self.data_files else get_data_patterns_locally(base_path)
```
This way, we are sure `split_modules` is never empty.

Reply:
> I think the validation of data_files={} should be elsewhere though. What do you think?

Yea indeed, probably in load_dataset_builder ?

> Maybe changing?

I think it's better if it raises an error rather than trying to make it run with data files that were not requested

Reply:
Feel free to merge then :)